agent.py

```diff
@@ -196,6 +206,266 @@ def _do_restart(self):
             self.collector.stop()
         sys.exit(AgentSupervisor.RESTART_EXIT_STATUS)
 
+def configcheck():
+    osname = get_os()
+    all_valid = True
+    for conf_path in glob.glob(os.path.join(get_confd_path(osname), "*.yaml")):
+        basename = os.path.basename(conf_path)
+        try:
+            check_yaml(conf_path)
+        except Exception, e:
+            all_valid = False
+            print "%s contains errors:\n    %s" % (basename, e)
+        else:
+            print "%s is valid" % basename
+    if all_valid:
+        print "All yaml files passed. You can now run the Datadog agent."
+        return 0
+    else:
+        print("Fix the invalid yaml files above in order to start the Datadog agent. "
+                "A useful external tool for yaml parsing can be found at "
+                "http://yaml-online-parser.appspot.com/")
+        return 1
+
+class Flare(object):
+    """
+    Compress all important logs and configuration files for debug,
+    and then send them to Datadog (which transfers them to Support)
+    """
+
+    DATADOG_SUPPORT_URL = '/zendesk/flare'
+    PASSWORD_REGEX = re.compile('( *(\w|_)*pass(word)?:).+')
+    COMMENT_REGEX = re.compile('^ *#.*')
+    APIKEY_REGEX = re.compile('^api_key:')
+
+    def __init__(self, cmdline=False, case_id=None):
+        self._case_id = case_id
+        self._cmdline = cmdline
+        self._init_tarfile()
+        self._save_logs_path(get_logging_config())
+        config = get_config()
+        self._api_key = config.get('api_key')
+        self._url = "{0}{1}".format(config.get('dd_url'), self.DATADOG_SUPPORT_URL)
+        self._hostname = get_hostname(config)
+        self._prefix = "datadog-{0}".format(self._hostname)
+
+    def collect(self):
+        if not self._api_key:
+            raise Exception('No api_key found')
+        self._print("Collecting logs and configuration files:")
+
+        self._add_logs_tar()
+        self._add_conf_tar()
+        self._print("  * datadog-agent configcheck output")
+        self._add_command_output_tar('configcheck.log', configcheck)
+        self._print("  * datadog-agent status output")
+        self._add_command_output_tar('status.log', self._supervisor_status)
+        self._print("  * datadog-agent info output")
+        self._add_command_output_tar('info.log', self._info_all)
+
+        self._print("Saving all files to {0}".format(self._tar_path))
+        self._tar.close()
+
+    # Upload the tar file
+    def upload(self, confirmation=True):
+        # Ask for confirmation first
+        if confirmation:
+            self._ask_for_confirmation()
+
+        email = self._ask_for_email()
+
+        self._print("Uploading {0} to Datadog Support".format(self._tar_path))
```

Comment:
The compression ratio is really good (only text files), on my tests 17MB -> 350KB, so it depends how huge it is. I could set an arbitrary limit on the size anyway.